crypto/fipsmodule/digest/digest.c

```diff
@@ -97,7 +99,9 @@ int EVP_MD_CTX_cleanup(EVP_MD_CTX *ctx) {
   OPENSSL_free(ctx->md_data);
 
   assert(ctx->pctx == NULL || ctx->pctx_ops != NULL);
-  if (ctx->pctx_ops) {
+  // pctx should be freed by the user of EVP_MD_CTX if
```

Comment:
When is this behaviour needed: when you call EVP_MD_CTX_cleanup(), but you expect it to leave behind `ctx->pctx`? What is expected from the `EVP_MD_CTX_init()` call that follows? Is it only resetting the digest operation? Will the same hash algorithm stay in the ctx, or that's also wiped? I think at least we need comments about the behaviour.
We also need test cases for it based on what we're expecting to happen.

Do we want to add service indicator tests to ensure that this manual setting of EVP_PKEY_CTX doesn't affect the `Approved` status of EVP_DigestSign/VerifyFinal?